File: Everything/Models/Lifting/LiftingWorkout.cs
using System;
using System.Collections.Generic;

namespace everything.Models
{
    public class LiftingWorkout : IForUser
    {
        public LiftingWorkout()
        {
            LiftSetLinks = new List<LiftSetLink>();
        }

        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Name { get; set; }
        public string Notes { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; }

        public virtual IList<LiftSetLink> LiftSetLinks { get; set; }
    }
}

File: Everything/ClientApp/src/models/lifting/LiftingWorkout.ts
export default class LiftingWorkout {
    id: number = 0;
    name: string | null = null;
    date: Date | null = null;
    notes: string = '';
    liftSetLinks: SimpleLiftSetLink[] = [];

    constructor(init?: Partial<LiftingWorkout>) {
        Object.assign(this, init);
    }
}

export class SimpleLiftSetLink {
    id: number = 0;
    name: string | null = null;

    constructor(init?: Partial<SimpleLiftSetLink>) {
        Object.assign(this, init);
    }
}